### PR TITLE
feat: 404 and 500 error pages

### LINKED ISSUE #171: Implement the 404 page
## Description

Implement the 404 page for when an identity does not exist and integrate it with error bondaries.

## Acceptance Criteria

- [x] Implement 404 page according to the design
- [x] Make use of React Error Boundaries

### PR DESCRIPTION
**Pull request by dominguesgm**
Implemented 404, 500 Error pages and Revoked Identity page.

This was implemented using Error Boundaries. In the instance of 404 pages, we had to target the iframe responsible for the CRA error overlay to hide it in development mode. This should be inactive by default in production environments.

Some restructuring was done to avoid replicating code between the 500 page and the boot error page.

NOTE: Revoke page does not integrate with `remove identity`, because this requires providing the mnemonic, which in turn should require a modal to pop up and request the mnemonic (this is not designed yet). **//cc** @andreforsousa @satazor 

Closes https://github.com/ipfs-shipyard/pm-idm/issues/171

### CODE REVIEW

File: src/app/App.css [hunk 1]
```diff
@@ -15,6 +15,7 @@
     & .page {
         width: calc(100% - 8rem);
         margin-left: 8rem;
+        display: flex;
```
Comment (satazor) [inline src/app/App.css:18]: Why did we need this?

File: src/app/error-boundary/ErrorBoundary.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.hideErrorOverlay > iframe {
```
Comment (satazor) [inline src/app/error-boundary/ErrorBoundary.css]: Do we have a more specific way to hide this. Is there an attribute or class in the iframe?

File: src/app/error-boundary/ErrorBoundary.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        }
+
+        return null;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hasError: false,
```
Comment (satazor) [inline src/app/error-boundary/ErrorBoundary.js]: I think we do not this need, we can just check against `error`.

File: src/app/error-boundary/ErrorBoundary.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        const { hasError, error } = this.state;
+
+        if (hasError) {
+            switch (error.code) {
+            case 'UNKNOWN_IDENTITY':
+                document.body.classList.add(styles.hideErrorOverlay);
+
+                return <Page404 />;
+            default:
+                document.body.classList.remove(styles.hideErrorOverlay);
```
Comment (satazor) [inline src/app/error-boundary/ErrorBoundary.js]: Don't we need to also remove the class if we don't have an error?

File: src/pages/errors/500/Page500.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Error from '../../../shared/components/error';
+
+const Page500 = (props) =>
+    (<Error
```
Comment (satazor) [inline src/pages/errors/500/Page500.js]: Nifty pick: can we have the parenthesis in the first line?

File: src/pages/errors/index.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import Page500 from './500';
+import Page404 from './404';
+
+export {
```
Comment (satazor) [inline src/pages/errors/index.js]: Can we simplify with just 2 exports?

File: src/shared/components/error/Error.css [hunk 7]
```diff
@@ -3,7 +3,6 @@
 
 .errorScreen {
     width: 100%;
-    height: 100%;
```
Comment (satazor) [inline src/shared/components/error/Error.css]: Why was this removed?

**Review by satazor — CHANGES_REQUESTED** [review]

File: src/app/App.css (hunk 1 above)
Comment (dominguesgm) [reply src/app/App.css:18]: We needed this because any child of the `.page` element cannot use `height: 100%`, to occupy the full viewport, given that `.page` has `flex: 1 0 auto` and thus no defined height

File: src/app/error-boundary/ErrorBoundary.css (hunk 2 above)
Comment (dominguesgm) [reply src/app/error-boundary/ErrorBoundary.css]: No, the iframe does not have any class name and the only attribute is inline styles.

I tried simply removing the iframe, but the `getDerivedStateFromProps` and `render` functions of `ErrorBoundary` run before the iframe is appended to the body, so that doesn't seem like a viable option.

File: src/app/error-boundary/ErrorBoundary.js (hunk 3 above)
Comment (dominguesgm) [reply src/app/error-boundary/ErrorBoundary.js]: I've removed it, thanks 👍

File: src/app/error-boundary/ErrorBoundary.js (hunk 4 above)
Comment (dominguesgm) [reply src/app/error-boundary/ErrorBoundary.js]: We can't to remove the class because the iframe is not removed from the DOM. Removing the class would reveal the iframe again, and thus should only happen when we run into an unexpected error.

File: src/pages/errors/500/Page500.js (hunk 5 above)
Comment (dominguesgm) [reply src/pages/errors/500/Page500.js]: Sure, I also prefer that way, nice catch 👍

File: src/pages/errors/index.js (hunk 6 above)
Comment (dominguesgm) [reply src/pages/errors/index.js]: Sure thing 👍

File: src/shared/components/error/Error.css (hunk 7 above)
Comment (dominguesgm) [reply src/shared/components/error/Error.css]: See the note on `display: flex` above. Both in the case where this `Error` component is used in `Boot` and the case where it is used in `App.page`, the parent component has an attribute `flex: 1 0 auto`, which means the `height: 100%` here does not affect it because the parent does not have a specified height

File: src/app/error-boundary/ErrorBoundary.css (hunk 2 above)
Comment (satazor) [reply src/app/error-boundary/ErrorBoundary.css]: Lets make it `.hideErrorOverlay > iframe[style]` to be a little more specific.

File: src/shared/components/error/Error.css (hunk 7 above)
Comment (satazor) [reply src/shared/components/error/Error.css]: Lets keep the height because we might not know if the parent is flexed.

File: src/shared/components/error/Error.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    state = {
+        showingDetails: false,
+    };
+
+    render() {
+        const { className, text, error, onRetry } = this.props;
+        const { showingDetails } = this.state;
+
+        return (
+            <div className={ classNames(styles.errorScreen, className) }>
```
Comment (satazor) [inline src/shared/components/error/Error.js]: Change the class name to `error`. Beware that the error class is already being used below, perhaps rename that one to `errorBlock`.

**Review by satazor — CHANGES_REQUESTED** [review]

File: src/shared/components/error/Error.css (hunk 7 above)
Comment (dominguesgm) [reply src/shared/components/error/Error.css]: 👍

File: src/shared/components/error/Error.js (hunk 8 above)
Comment (dominguesgm) [reply src/shared/components/error/Error.js]: 👍

File: src/app/error-boundary/ErrorBoundary.css (hunk 2 above)
Comment (dominguesgm) [reply src/app/error-boundary/ErrorBoundary.css]: 👍
